### PR TITLE
Mantine TimeInput

### PR DESCRIPTION
**Pull request by ranquild**
Follow the same approach used by the mantine's `DateInput`:
- Allows controlled and uncontrolled usage
- Accepts and return a `Date`
- Resets the input value on blur

`TextInput` is used internally so no custom styles are needed.

How to test:
- `yarn storybook`
- Open http://localhost:6006/?path=/docs/inputs-timeinput--default-story
- To test AM/PM, change your Clock settings in the OS settings, and re-open the browser

Screenshot 2023-10-13 at 14 29 53
Screenshot 2023-10-18 at 3 39 45 PM
Screenshot 2023-10-18 at 3 39 55 PM

### CODE REVIEW

File: frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { TimeInput as MantineTimeInput } from "@mantine/dates";
+import type { TimeInputProps as MantineTimeInputProps } from "@mantine/dates";
+
+const TIME_FORMAT = "HH:mm";
+
+export type TimeInputProps = Omit<
+  MantineTimeInputProps,
+  "value" | "defaultValue" | "onChange"
+> & {
+  value?: Date | null;
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx:13]: `null` is allowed to be compatible with `DateInput`. It has `clearable` option (`false` by default).

File: frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  defaultValue?: Date | null;
+  onChange?: (value: Date) => void;
+};
+
+export function TimeInput({
+  value,
+  defaultValue = value,
+  onChange,
+  ...props
+}: TimeInputProps) {
```
Comment (kulyk) [inline frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx:23]: Curios why we need `value` and `defaultValue`? Can't we do something like that:

```jsx
function TimeInput({ value, onChange }) {
  const inputValue = value ? formatTime(value) : ""
  
  const handleChange = event => {
    const newTime = parseTime(event.target.value)
    onChange(newTime)
  }

  return <MantineTimeInput value={inputValue} onChange={handleChange} />
}
```

Comment (ranquild) [reply frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx:23]: `defalueValue` is needed for uncontrolled usage (all mantine components support it)

Comment (ranquild) [reply frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx:23]: e.g. `<TimeInput defaultValue={new Date()} />` - no `onChange` handler

Comment (kulyk) [reply frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx:23]: Do we need uncontrolled inputs though?

Comment (ranquild) [reply frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx:23]: all current mantine inputs support uncontrolled usage

Comment (kulyk) [reply frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx:23]: Oh, I see, makes sense to keep things consistent

**Review by kulyk — APPROVED** [review]

**Review by kdoh — APPROVED** [review]

**Pull request merged**